aptos-move/framework/src/natives/cryptography/groups/blst_backend.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::VecDeque;
+use better_any::{Tid, TidAble};
+use blst::{blst_fp12, blst_fr, blst_p1, blst_p2, blst_p2_affine, blst_p2_affine_compress, blst_p2_to_affine};
+use move_binary_format::errors::PartialVMResult;
+use move_core_types::gas_algebra::{InternalGas, InternalGasPerArg, NumArgs};
+use move_vm_runtime::native_functions::NativeContext;
+use move_vm_types::loaded_data::runtime_types::Type;
+use move_vm_types::natives::function::NativeResult;
+use move_vm_types::pop_arg;
+use move_vm_types::values::Value;
+use smallvec::smallvec;
+use crate::natives::cryptography::groups::abort_codes::E_UNKNOWN_GROUP;
+
+#[derive(Debug, Clone)]
+pub struct Bls12381GasParameters {
+    pub fp12_to_bendian: InternalGasPerArg,
+    pub p2_affine_compress: InternalGasPerArg,
+    pub pairing_product_base: InternalGas,
+    pub pairing_product_per_pair: InternalGasPerArg,
+}
+
+#[derive(Debug, Clone)]
+pub struct GasParameters {
+    pub bls12_381: Bls12381GasParameters,
```

Comment:
Why do you have this layer of indirection? Could've declared everything here directly, no?

Reply:
We need to support other curves in the future.